TShockAPI/Bouncer.cs

```diff
@@ -2496,5 +2511,45 @@ internal static int GetMaxPlaceStyle(int tileID)
 			{ ProjectileID.GoldShortswordStab, ItemID.GoldShortsword },
 			{ ProjectileID.PlatinumShortswordStab, ItemID.PlatinumShortsword }
 		};
+
+		private Dictionary<short, short> Projectile_MinValuesAI = new Dictionary<short, short> {
+			{ 611, -1 },
+
+			{ 950, 0 }
+		};
+		private Dictionary<short, short> Projectile_MaxValuesAI = new Dictionary<short, short> {
+			{ 611, 1 },
+
+			{ 950, 0 }
+		};
+
+		private Dictionary<short, short> Projectile_MinValuesAI2 = new Dictionary<short, short> {
+			{ 405, 0 },
+			{ 410, 0 },
+
+			{ 424, 0 },
+			{ 425, 0 },
+			{ 426, 0 },
+
+			{ 612, 0 },
+			{ 953, 0 },
+
+			{ 756, 0 },
+			{ 522, 0 }
+		};
+		private Dictionary<short, short> Projectile_MaxValuesAI2 = new Dictionary<short, short> {
+			{ 405, 2 },
+			{ 410, 2 },
+
+			{ 424, 1 },
+			{ 425, 1 },
+			{ 426, 1 },
+
+			{ 612, 2 },
+			{ 953, 2 },
+
+			{ 756, 1 },
+			{ 522, 0 }
```

Comment:
```suggestion
			{ 522, 40f }
```
Source: `Projectile.AI()`, `if (this.ai[1] > 40f) { Kill(); }`